### PR TITLE
replace dbh with dbic

### PR DESCRIPTION
**Pull request by chylli**

### CODE REVIEW

File: lib/Data/Chronicle/Reader.pm [hunk 1]
```diff
@@ -140,11 +152,13 @@ sub get_for {
...
     my $db_timestamp = Date::Utility->new($date_for)->db_timestamp;
 
-    die "Requesting for historical data without a valid DB connection [$category,$name,$date_for]" if not defined $self->db_handle;
+    die "Requesting for historical data without a valid DB connection [$category,$name,$date_for]" if not defined $self->dbic;
 
-    my $db_data =
-        $self->db_handle->selectall_hashref(q{SELECT * FROM chronicle where category=? and name=? and timestamp<=? order by timestamp desc limit 1},
-        'id', {}, $category, $name, $db_timestamp);
+    my $db_data = $self->dbic->run(
+        ping => sub {
```
Comment (torsten-deriv) [inline lib/Data/Chronicle/Reader.pm]: why `ping`? Why not `fixup`?

File: lib/Data/Chronicle/Reader.pm [hunk 2]
```diff
@@ -170,12 +184,13 @@ sub get_for_period {
...
 
-    die "Requesting for historical period data without a valid DB connection [$category,$name]" if not defined $self->db_handle;
+    die "Requesting for historical period data without a valid DB connection [$category,$name]" if not defined $self->dbic;
 
-    my $db_data =
-        $self->db_handle->selectall_hashref(
-        q{SELECT * FROM chronicle where category=? and name=? and timestamp<=? AND timestamp >=? order by timestamp desc},
-        'id', {}, $category, $name, $end_timestamp, $start_timestamp);
+    my $db_data = $self->dbic->run(
+        ping => sub {
```
Comment (torsten-deriv) [inline lib/Data/Chronicle/Reader.pm]: why `ping`?

File: lib/Data/Chronicle/Writer.pm [hunk 3]
```diff
@@ -162,7 +168,9 @@ sub _archive {
 
     my $db_timestamp = $rec_date->db_timestamp;
 
-    return $self->db_handle->prepare(<<'SQL')->execute($category, $name, $value, $db_timestamp);
+    return $self->dbic->run(
+        ping => sub {
```
Comment (torsten-deriv) [inline lib/Data/Chronicle/Writer.pm]: this can probably also be run with `fixup`.

**Comment by chylli** [comment]
@tfoertsch Fixed.

**Pull request merged**